src/script_opt/ZAM/Support.cc

```diff
@@ -117,6 +117,12 @@ bool file_mgr_set_reassembly_buffer(StringVal* file_id, uint64_t max) {
 bool ZAM_error = false;
 
 bool is_ZAM_compilable(const ProfileFunc* pf, const char** reason) {
+    if ( has_AST_node_unknown_to_script_opt(pf, true) ) {
+        if ( reason )
+            *reason = "unknown AST node type";
```

Comment:
Might make sense to get the offending node type out of the helper (or change to std::pair<bool, reason: string>`) so that the message provides a hint which stmt/expr it was which seems important.

Reply:
I see the benefit. Here I wasn't looking to open up that API, just to add an element to its functionality.

Reply:
Oh, I just tried and now see const `char** reason` as output parameter which would require memory management for a formatted string. Resolving.